src/pgduckdb_ddl.cpp

```diff
@@ -282,7 +305,18 @@ DuckdbHandleDDL(PlannedStmt *pstmt, const char *query_string, ParamListInfo para
 
 		Query *rewritten_query_copy = (Query *)copyObjectImpl(query);
 
+		/*
+		 * We temporarily set the force_execution GUC to true, so that we can
+		 * plan the query in DuckDB. This is needed for a CTAS into a DuckDB
+		 * table, where the query itself does not require DuckDB execution.
+		 * If we don't set force_execution to true, the query will be planned
+		 * using Postgres, but then later executed using DuckDB when we
+		 * actually write into the DuckDB table.
+		 */
+		auto save_nestlevel = NewGUCNestLevel();
+		SetConfigOption("duckdb.force_execution", "true", PGC_USERSET, PGC_S_SESSION);
```

Comment:
Do we have to go through the GUC? Why not using a different "force_execution" that would override it? Is it to limit the changes in this PR?

Reply:
Fixed now by calling `DuckdbPlanNode` directly.